macrosynergy/pnl/notional_positions.py

```diff
@@ -65,6 +65,94 @@ def _apply_slip(
         )
 
 
+def _check_df_for_contract_signals(
+    df: QuantamentalDataFrame,
+    sname: str,
+    fids: List[str],
+) -> None:
+    """
+    Checks if the dataframe contains contract signals for the specified strategy
+    and the specified contract identifiers.
+
+    :param <QuantamentalDataFrame> df: Quantamental dataframe with contract signals and returns.
+    :param <str> sname: the name of the strategy.
+    :param <List[str]> fids: list of contract identifiers to apply the slip to.
+    """
+    assert isinstance(df, QuantamentalDataFrame)
+    assert isinstance(sname, str)
+    assert (
+        isinstance(fids, list)
+        and len(fids) > 0
+        and all([isinstance(x, str) for x in fids])
+    )
+
+    sig_ident: str = f"_CSIG_{sname}"
+    _check_conts: Set = set([f"{contx}{sig_ident}" for contx in fids])
+    _found_conts: Set = set((df["cid"] + "_" + df["xcat"]).unique())
+    if not _check_conts.issubset(_found_conts):
+        raise ValueError(
+            f"Contract signals for all contracts not in dataframe. \n"
+            f"Missing: {_check_conts - _found_conts}"
+        )
+
+    return
+
+
+def _vol_target_positions(
+    df: pd.DataFrame,
+    sname: str,
+    fids: List[str],
+    dollar_per_signal: Numeric = 1.0,
+    vol_target: Numeric = 0.1,
```

Comment:
Minor point, but unless I am mistaken, we normally refer to volatility is percent, so `10` and not `0.1`.

Reply:
Fixed